### PR TITLE
[onert] Use TFLite 2.8 for test driver

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit updates test drivers and related libs to use TFLite 2.8 instead of TFLite 1.13.1. 
For easy update, "libs/tflite" uses TFLite C API instead of internal headers.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: https://github.com/Samsung/ONE/issues/5930
Draft: #9671

### CODE REVIEW

File: tests/scripts/list/tflite_comparator.x86_64.cpu.list [hunk 1]
```diff
@@ -11,7 +10,7 @@ logistic
 max_pool_2d
 mean
 reduce_max
-reduce_sum
+reduce_sum/float
```
Comment (hseok-oh) [inline tests/scripts/list/tflite_comparator.x86_64.cpu.list:13]: `REDUCE_SUM` uint8 test fail on CPU backend.

**Comment by glistening** [comment]
I will start to review. But it will take time. Change is big. : )

At a glance, it consists of the followings:

1) Using tflite API
2) Remove `tflite_vanilar_run` ( I don't know why and history about this. )
3) Update version
4) Filter out some test case like reduceSum uint8
(maybe more?)

**Comment by hseok-oh** [comment]
>At a glance, it consists of the followings:

You're right.

>Remove tflite_vanilar_run ( I don't know why and history about this. )

`tflite_vanilla_run` is temporary test driver to use TFLite 2.x. It is copy of `tflite_run` with some fix to use updated headers. By this PR, `tflite_run` will use TFLite 2.8, so we don't need `tflite_vanilla_run` any more.

File: tests/tools/tflite_run/src/tensor_dumper.cc [hunk 2]
```diff
@@ -49,17 +64,30 @@ void TensorDumper::dump(const std::string &filename) const
   std::ofstream file(filename, std::ios::out | std::ios::binary);
 
   // Write number of tensors
-  uint32_t num_tensors = static_cast<uint32_t>(_tensors.size());
+  uint32_t num_tensors =
+    static_cast<uint32_t>(_input_tensors.size()) + static_cast<uint32_t>(_input_tensors.size());
```
Comment (glistening) [inline tests/tools/tflite_run/src/tensor_dumper.cc:68]:
```suggestion
  uint32_t num_tensors =
    static_cast<uint32_t>(_input_tensors.size()) + static_cast<uint32_t>(_output_tensors.size());
```

File: tests/tools/tflite_run/src/tensor_loader.h [hunk 3]
```diff
@@ -35,17 +36,19 @@ namespace TFLiteRun
...
 public:
-  TensorLoader(tflite::Interpreter &interpreter);
+  TensorLoader(TfLiteInterpreter &interpreter);
   void loadDumpedTensors(const std::string &filename);
-  void loadRawTensors(const std::string &filename, const std::vector<int> &tensor_indices);
-  const nnfw::tflite::TensorView<float> &get(int tensor_idx) const;
-  size_t getNums() const { return _tensor_map.size(); }
+  void loadRawInputTensors(const std::string &filename);
+  const nnfw::tflite::TensorView<float> &getOutput(int tensor_idx) const;
+  // size_t getNums() const { return _tensor_map.size(); }
```
Comment (glistening) [inline tests/tools/tflite_run/src/tensor_loader.h:43]:
```suggestion
```

**Review by glistening — COMMENTED** [review]
- Please fix typos `obtainined`. Replacing would be faster. : )
- Please see my other comments.
- It looks good except for above.

**Comment by hseok-oh** [comment]
I've updated. PTAL

**Review by chunseoklee — APPROVED** [review]

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**